src/frontend/Ast_to_Mir.ml

```diff
@@ -387,22 +419,37 @@ let check_sizedtype name st =
     | SArray (t, s) ->
         let e = trans_expr s in
         let ll, t = sizedtype t in
-        (check s e @ ll, SizedType.SArray (t, e)) in
+        (check s e @ ll, SizedType.SArray (t, e))
+    | STuple ts ->
+        let checks, ts = List.unzip (List.map ~f:sizedtype ts) in
+        (List.concat checks, STuple ts) in
   let ll, st = sizedtype st in
   (ll, Type.Sized st)
 
-let trans_decl {transform_action; dadlevel} smeta
-    (decl_type : Ast.typed_expression SizedType.t) transform identifier
+(* The statements that constrain and check a variable, given its context *)
+let var_constrain_check_stmts dconstrain loc adlevel decl_id decl_var trans
+    type_ =
+  match (dconstrain, type_) with
+  | (Some Constrain | Some Unconstrain), Type.Sized _ ->
+      check_transform_shape decl_id decl_var loc trans
+  | Some Check, Type.Sized st ->
+      check_transform_shape decl_id decl_var loc trans
+      @ check_decl decl_var st trans loc adlevel
+  | _ -> []
```

Comment:
[optional] imo if we can list these out then it would be nice so that in the future if something gets added here the compiler fails at this match

```ocaml
  | (Some IgnoreTransform | None), Sized _-> []
  | (Some Constrain | Some Unconstrain| Some Check | Some IgnoreTransform| None), Type.Unsized _ -> []
```